src/MusicTheory/Music.elm

```diff
@@ -0,0 +1,325 @@
+module MusicTheory.Music exposing
+    ( Control(..)
+    , Music(..)
+    , aFlat
+    , aNatural
+    , aSharp
+    , addDuration
+    , bFlat
+    , bNatural
+    , bSharp
+    , cFlat
+    , cNatural
+    , cSharp
+    , chord
+    , dFlat
+    , dNatural
+    , dSharp
+    , eFlat
+    , eNatural
+    , eSharp
+    , eighthNoteTriplet
+    , fFlat
+    , fNatural
+    , fSharp
+    , gFlat
+    , gNatural
+    , gSharp
+    , key
+    , line
+    , map
+    , modify
+    , note
+    , pMap
+    , pToList
+    , par
+    , quarterNoteTriplet
+    , rest
+    , seq
+    , tempo
+    , timeSignature
+    , times
+    , toList
+    )
+
+import MusicTheory.Duration as Duration exposing (Duration)
+import MusicTheory.Key exposing (Key)
+import MusicTheory.Letter exposing (Letter(..))
+import MusicTheory.Octave exposing (Octave)
+import MusicTheory.Pitch as Pitch exposing (Pitch)
+import MusicTheory.Tuplet as Tuplet exposing (Tuplet)
+
+
+
+-- TYPES
+
+
+type alias Primitive a =
+    ( Duration, List a )
+
+
+type alias Bpm =
+    Float
+
+
+type Control
+    = Tuplet Tuplet
+    | KeySignature Key
+    | TimeSignature Int Int
+    | Tempo Duration Bpm
+
+
+type Music a
+    = Prim (Primitive a)
+    | Seq (Music a) (Music a)
+    | Par (Music a) (Music a)
+    | Modify Control (Music a)
+
+
+
+-- CONSTRUCTORS
+
+
+note : Duration -> a -> Music a
+note duration a =
+    Prim <| ( duration, [ a ] )
+
+
+rest : Duration -> Music a
+rest duration =
+    Prim <| ( duration, [] )
+
+
+seq : Music a -> Music a -> Music a
+seq =
+    Seq
+
+
+par : Music a -> Music a -> Music a
+par =
+    Par
+
+
+key : Key -> Music a -> Music a
+key k =
+    modify (KeySignature k)
+
+
+timeSignature : Int -> Int -> Music a -> Music a
+timeSignature beats duration =
+    modify (TimeSignature beats duration)
+
+
+tempo : Duration -> Float -> Music a -> Music a
+tempo d t =
+    modify <| Tempo d t
+
+
+modify : Control -> Music a -> Music a
+modify =
+    Modify
+
+
+quarterNoteTriplet : (Duration -> Music a) -> (Duration -> Music a) -> (Duration -> Music a) -> Music a
+quarterNoteTriplet m1 m2 m3 =
+    modify (Tuplet (Tuplet.triplet Tuplet.Quarter)) (line [ m1 Duration.quarterNote, m2 Duration.quarterNote, m3 Duration.quarterNote ])
+
+
+eighthNoteTriplet : (Duration -> Music a) -> (Duration -> Music a) -> (Duration -> Music a) -> Music a
```

Comment:
This could be a constructor for tuplets.

The things that can be passed to this function are the note constructors applied to an octave or chords.

So it is not possible (very easily) to create tuplets that contain `Music a` values with an invalid duration.

The usage can be seen in the test, e.g. here: https://github.com/elm-music/music-theory/pull/35/files#diff-4345091e9177eabbdf790cdff1e2e15fR146